src/write_gcm.c

```diff
@@ -2367,11 +2372,12 @@ static int wg_write(const data_set_t *ds, const value_list_t *vl,
   pthread_mutex_lock(&queue->mutex);
   // Backpressure. If queue is backed up then something has gone horribly wrong.
   // Maybe the queue processor died.
-  size_t abnormal_limit = 5 * MAX_QUEUE_SIZE;
-  if (queue->size > abnormal_limit) {
+  if (queue->size > QUEUE_DROP_SIZE) {
+    DEBUG("write_gcm: Dropping data point because queue has size %zd",
```

Comment:
Two things:


   1. As a policy matter, I'm happy to go either way. When the queue fills
   up like that, maybe indeed the right thing to do is flush the whole thing
   and start afresh. So I have no objection to doing it that way if you want.
   We could even have such flush behavior controlled by a config flag if we
   want to try it.
   2. But to address the point behind your question, I don't think a "bad
   point" can hold up the queue. The way the consumer thread works is (after
   the queue reaches a certain size or after the 1 minute timeout) to wake up,
   take ownership of the entire queue as it exists currently, and then start
   feeding that upstream. If some point in that queue has errors the point
   will just be discarded.. (There's no retry)


On Thu, Oct 22, 2015 at 11:09 AM, Jeremy Katz <notifications@github.com>
wrote:

> In src/write_gcm.c
> <https://github.com/Stackdriver/collectd/pull/26#discussion_r42760480>:
>
> > @@ -2367,11 +2372,12 @@ static int wg_write(const data_set_t *ds, const value_list_t *vl,
> >    pthread_mutex_lock(&queue->mutex);
> >    // Backpressure. If queue is backed up then something has gone horribly wrong.
> >    // Maybe the queue processor died.
> > -  size_t abnormal_limit = 5 * MAX_QUEUE_SIZE;
> > -  if (queue->size > abnormal_limit) {
> > +  if (queue->size > QUEUE_DROP_SIZE) {
> > +    DEBUG("write_gcm: Dropping data point because queue has size %zd",
>
> Should we drop the oldest point rather than the newest point in these
> situations? Maybe there's something "bad" about the old point and that's
> why they're there?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Stackdriver/collectd/pull/26/files#r42760480>.
>